File: server/src/main/java/com/im/nettychat/protocol/request/RegisterRequest.java
/*
 * Project: com.im.nettychat.protocol.request
 * 
 * File Created at 2018/12/20
 * 
 * Copyright 2018 CMCC Corporation Limited.
 * All rights reserved.
 *
 * This software is the confidential and proprietary information of
 * ZYHY Company. ("Confidential Information").  You shall not
 * disclose such Confidential Information and shall use it only in
 * accordance with the terms of the license.
 */
package com.im.nettychat.protocol.request;

import com.im.nettychat.common.Command;
import com.im.nettychat.protocol.RequestPacket;
import lombok.Data;

/**
 * @author hejianglong
 * @Desc
 * @date 2018/12/20 下午9:33
 */
@Data
public class RegisterRequest extends RequestPacket {

    private String name;

    private String username;

    private String password;

    @Override
    public Byte getCommand() {
        return Command.REGISTER;
    }
}

File: AndroidClient/app/src/main/java/com/chat/androidclient/mvvm/model/RegisterRequest.kt
package com.chat.androidclient.mvvm.model

import com.chat.androidclient.mvvm.procotol.Packet
import com.chat.androidclient.network.NeetClient

/**
 * Created by lps on 2018/12/25 10:27.
 */
class RegisterRequest(var name:String,var username:String, var password:String): Packet() {
    override val command: Byte
        get() = Command.REGISTER
    
    override fun toString(): String {
        return "RegisterRequest(name='$name', username='$username', password='$password')"
    }
    
}


